PSPDFKitPlugin/PSPDFKitPlugin.m

```diff
@@ -423,16 +395,67 @@ - (BOOL)isImagePath:(NSString *)path {
     return [pathExtension isEqualToString:@"png"] || [pathExtension isEqualToString:@"jpeg"] || [pathExtension isEqualToString:@"jpg"];
 }
 
-- (NSInteger)enumValueForKey:(NSString *)key ofType:(NSString *)type withDefault:(int)defaultValue
-{
+- (void)createXFDFDocumentWithPath:(NSString *)path {
+    NSURL *xfdfFileURL;
+    if (path.isAbsolutePath) {
+        xfdfFileURL = [self pdfFileURLWithPath:path];
+    } else {
+        // Create the XFDF file in the ~/Documents directory
+        NSString *docsFolder = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES).firstObject;
+        xfdfFileURL = [NSURL fileURLWithPath:[docsFolder stringByAppendingPathComponent:path]];
+    }
+
+    // Create an XFDF file from the current document if one doesn't already exist.
+    if (![NSFileManager.defaultManager fileExistsAtPath:(NSString *)xfdfFileURL.path]) {
+        // Create the folder where the XFDF file will be saved.
+        NSError *createFolderError;
+        if (![[NSFileManager defaultManager] createDirectoryAtPath:[xfdfFileURL.path stringByDeletingLastPathComponent] withIntermediateDirectories:YES attributes:nil error:&createFolderError]) {
+            NSLog(@"Failed to create directory: %@", createFolderError.localizedDescription);
+            return;
+        }
+
+        // Collect all existing annotations from the document
+        NSMutableArray *annotations = [NSMutableArray array];
+        for (NSArray *pageAnnots in [_pdfDocument allAnnotationsOfType:PSPDFAnnotationTypeAll].allValues) {
+            [annotations addObjectsFromArray:pageAnnots];
+        }
```

Comment:
Is it always expected and desired to store existing annotations in XFDF?
I would think only new annotations are stored there.
What do you think?

Reply:
I see that existing annotations wouldn't be displayed in that case, since we don't add the file annotation provider by default.

Maybe we could make moving the existing annotations from a document to XFDF a separate method, so customers can decide on the behavior themselves?

Reply:
Good point about the automagic! I'll remove this logic for now.

Reply:
Done in https://github.com/PSPDFKit/Cordova-iOS/pull/75/commits/42d810f3082f0dec08985931ab06702f44d22621